scadalts-ui/src/views/DataObjects/Scripting/index.vue

```diff
@@ -1,33 +1,420 @@
+			this.searchFilters.page = data.page;
+			this.searchFilters.itemsPerPage = data.itemsPerPage;
+			this.searchFilters.sortBy = data.sortBy;
+			this.searchFilters.sortDesc = data.sortDesc;
+			this.fetchScriptList();
+		},
+	},
 	data() {
 		return {
-			title: 'Scripting',
-			url: 'scripting.shtm',			
-		}
+			xidUnique: true,
+			ruleNotNull: (v) => !!v || this.$t('validation.rule.notNull'),
+			ruleMaxLen40: (v) => !!(v.length < 40) || this.$t('validation.rule.maxLen')+40,
+			ruleMaxLen50: (v) => !!(v.length < 50) || this.$t('validation.rule.maxLen')+50,
+			ruleXidUnique: (v) => !!this.xidUnique || this.$t('validation.rule.xid.notUnique'),
+			snackbarMessage: '',
+			snackbar: false,
+			dialog: false,
+			search: '',
+			scriptListFiltered: [],
+			datapointToSave: null,
+			selectedDatapointId: null,
+			datapoints: [],
+			options: {},
+			selectedScriptId: -1,
+			selectedScript: null,
+			scriptForm: {
+				id: null,
+				name: '',
+				xid: '',
+				pointsOnContext: [],
+				datasourceContext: '',
+				datapointContext: '',
+				script: '',
+				type: 'CONTEXTUALIZED_SCRIPT',
+				typeKey: 'event.audit.scripts',
+				new: false,
+			},
+			searchFilters: {
+				keywordfs: '',
+				page: 1,
+				itemsPerPage: 10,
+				sortBy: [],
+				sortDesc: [],
+			},
+			totalScripts: 100,
+			scriptList: [],
+			loading: false,
+			headers: [
+				{
+					text: this.$t('scriptList.id'),
+					sortable: true,
+					align: 'center',
+					value: 'id',
+				},
+				{
+					text: this.$t('scriptList.xid'),
+					align: 'center',
+					sortable: true,
+					value: 'xid',
+				},
+				{
+					text: this.$t('scriptList.name'),
+					align: 'center',
+					sortable: true,
+					value: 'name',
+				},
+				{
+					text: this.$t('scriptList.actions'),
+					align: 'center',
+					sortable: true,
+					value: 'actions',
+				},
+			],
+			sortModeOptions: [
+				{ label: 'Ascending', value: 'ASC' },
+				{ label: 'Descending', value: 'DEC' },
+			],
+		};
 	},
-
 	computed: {
-		location() {
-			return this.convertBrowserPath(this.url);
-		}
-	}    
-}
-</script>
\ No newline at end of file
+		filteredDatapoints() {
+			const p = this.datapoints.find((x) => x.id === this.selectedDatapointId);
+			return this.datapoints
+			.filter(dp => this.scriptForm.pointsOnContext
+				.filter(sp => dp.xid === sp.dataPointXid)
+			.length === 0)
+		},
+	},
+
+	methods: {
+		async validateXid() {
+			const response = await this.$store.dispatch('validateXid', this.scriptForm)
+			this.xidUnique = response.xidRepeated === 'false' ? true: false;
+			this.$refs.xidInput.validate();
+		},
+		createNewScript() {
+			this.selectedScriptId = -1;
+
+			this.selectedScript = null;
+			this.scriptForm.id = -1;
+			this.scriptForm.xid = '';
+			this.scriptForm.name = '';
+			this.scriptForm.pointsOnContext = [];
+			this.scriptForm.script = '';
+			this.scriptForm.datasourceContext = 'ds';
+			this.scriptForm.datapointContext = 'dp';
+
+			this.$refs.editForm.resetValidation()
```

Comment:
This `resetValidation()` method do not work when the "scriptForm" is not initialized. Try to add validation to not pass the "undefined" reference to this method execution.

![image](https://user-images.githubusercontent.com/22638815/151140592-29a41764-d9e8-48b1-b439-c37b50c8b2de.png)

Reply:
Fixed in this [commit](https://github.com/SCADA-LTS/Scada-LTS/pull/1980/commits/5c13cc97cfae4f74ecef5099063d593ea60aa9c4)